base/sparse/linalg.jl

```diff
@@ -585,42 +587,44 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
         S_old = copy(S)
         for j = 1:t
             for i = 1:n
-                S[i,j] = Y[i,j]==0?1:sign(Y[i,j])
+                S[i,j] = Y[i,j]==0?one(Y[i,j]):sign(Y[i,j])
```

Comment:
I'd rather put  `one(sign(Y[i,j]))`, to be consistent with the definition of  `sign(::Complex)`.